### PR TITLE
Remove unneeded namespacing in header files

### LINKED ISSUE #857: Some headers #include other 'mbedTLS' headers as "mbedtls/*.h" rather than "*.h"
Dear 'mbedTLS' developers,

While embedding 'mbedTLS' (ver 2.4.2) sources into our source tree we noticed that some headers include other mbedTLS headers as:  _#include "**mbedtls**/foo.h"_ (rather than just: _#include "foo.h"_). It happens but in just a handful of cases (listed below), everywhere else it uses a more manageable: _#include "foo.h"_.

While it's no big deal it looks like an accidental oversight which makes 'mbedTLS' just a tad less a "drop-in"; and, it looks like an easy formal fix. -- Can you please take a look?

```
cmac.h:27:#include "mbedtls/cipher.h"
config.h:2589://#define YOTTA_CFG_MBEDTLS_TARGET_CONFIG_FILE "mbedtls/target_config.h"
ctr_drbg.h:29:#include "mbedtls/threading.h"
hmac_drbg.h:29:#include "mbedtls/threading.h"
net.h:27:#include "mbedtls/net_sockets.h"
platform.h:33:#include "mbedtls/platform_time.h"
ssl.h:55:#include "mbedtls/platform_time.h"
```

Thanks much for the great embeddable product!
(We have recently switched to it from GnuTLS which was much harder to manage due to a lot of 3rd-party DLL dependencies.)

### PR DESCRIPTION
**Pull request by RonEld**
Remove the `mbedtls` namesapcing in the `#include` in header files
Resolves #857

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,5 +1,11 @@
 mbed TLS ChangeLog (Sorted per branch, date)
 
+= mbed TLS x.x.x branch released xxxx-xx-xx
+
+Bugfix
+   * Fix namespacing in header files. REmove the `mbedtls` namespacing in
```
Comment (hanno-arm) [inline ChangeLog]: Typo

**Comment by RonEld** [comment]
@hanno-arm Thank you for your review
I fixed the typo. Please approve

File: include/mbedtls/net.h [hunk 2]
```diff
@@ -20,11 +20,11 @@
...
  *
- * \deprecated Superseded by mbedtls/net_sockets.h
+ * \deprecated Superseded by net_sockets.h
  */
 
 #if !defined(MBEDTLS_DEPRECATED_REMOVED)
-#include "mbedtls/net_sockets.h"
+#include "net_sockets.h"
 #if defined(MBEDTLS_DEPRECATED_WARNING)
 #warning "Deprecated header file: Superseded by mbedtls/net_sockets.h"
```
Comment (andresag01) [inline include/mbedtls/net.h]: Should we change this one as well?

Comment (RonEld) [reply include/mbedtls/net.h]: @andresag01 Why do you think it shouldn't change? Although `net.h` is superseded, it can still be included.

Comment (andresag01) [reply include/mbedtls/net.h]: This was a long time ago, so what I probably meant is that in the message for this warning there is a `mbedtls/net_sockets.h`. Do we want to make that `net_sockets.h` as well? Its a really minor thing and certainly not a stopper, more of a question. I will accept the PR.

Comment (RonEld) [reply include/mbedtls/net.h]: I think in the message, it's more appropriate to keep it as is, as it is intended for human readers, and it's informative with the current wording

**Comment by andresag01** [comment]
@RonEld: Thanks for looking into this again, but could you please rebase instead of creating a merge commit. For example, the diff now shows as if you had introduced a new config file and in general its not ideal to have these merge commits in the history.

**Comment by RonEld** [comment]
@andresag01 I have removed the merge commit and rebased.
Please review

**Comment by RonEld** [comment]
backports created in #1779 and #1780 for mbedtls-2.1 and mbedtls-2.7 respectively

File: include/mbedtls/net.h [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
  * \file net.h
  *
- * \brief Deprecated header file that includes mbedtls/net_sockets.h
+ * \brief Deprecated header file that includes net_sockets.h
```
Comment (andresag01) [inline include/mbedtls/net.h]: Minor: Why did you remove `mbedtls/` from this comment for example and not from the one below?

File: configs/config-ccm-psk-tls1_2.h [hunk 4]
```diff
@@ -83,6 +83,6 @@
  */
 #define MBEDTLS_SSL_MAX_CONTENT_LEN             512
 
-#include "mbedtls/check_config.h"
+#include "check_config.h"
```
Comment (andresag01) [inline configs/config-ccm-psk-tls1_2.h]: The README.txt file in the configs directory states that there are two options to use these config files
>1. Replace the default file include/mbedtls/config.h with the chosen one.
   (Depending on your compiler, you may need to adjust the line with
   #include "mbedtls/check_config.h" then.)
>
>2. Define MBEDTLS_CONFIG_FILE and adjust the include path accordingly.

So in case (1) the user has to do nothing and just copy the file over as stated in the readme (actually, I think the information in brackets is no longer relevant). However, in the case of (2) the user will probably need to add `./configs` to the include path as well as `./include/mbedtls` (because this patch removes the namespacing bit). Also, note that without this change, case (1) will also work with the `mbedtls/` prefix, its just not ideal and its already properly documented.

So I am not too sure that we should remove the namespacing from these config files. But if we do want to remove, we should at least fix the documentation accordingly.

File: include/mbedtls/net.h (hunk 3 above)
Comment (RonEld) [reply include/mbedtls/net.h]: I don't really have a good answer.
Probably because in this comment, it's more reasonable that a human reader will read the file within the same folder, and in the warning message, the compilation will probably be from the root folder. I'm fine either way

**Comment by RonEld** [comment]
Added the "needs backports" label, as the backports are still pending reviews

File: configs/config-ccm-psk-tls1_2.h (hunk 4 above)
Comment (RonEld) [reply configs/config-ccm-psk-tls1_2.h]: I'll revert the changes in the configs

**Comment by RonEld** [comment]
@andresag01 I have reverted the changes in the `configs` folder, and added an `mbedtls` namespace in the `config-no-entropy.h` file as well. Please review

cc @hanno-arm

File: configs/config-no-entropy.h [hunk 5]
```diff
@@ -87,6 +87,6 @@
 /* Miscellaneous options */
 #define MBEDTLS_AES_ROM_TABLES
 
-#include "check_config.h"
+#include "mbedtls/check_config.h"
```
Comment (andresag01) [inline configs/config-no-entropy.h]: I think this might not be ideal. The issue is that this file is copied by some import script directly into the inc/mbedtls directory in Mbed TLS, so technically the `mbedtls/` prefix is not needed.

**Comment by RonEld** [comment]
@andresag01 I have removed the `mbedtls` namespacing as you requested.
I will update the backports once this PR is approved

File: include/mbedtls/net.h (hunk 2 above)
Comment (sbutcher-arm) [reply include/mbedtls/net.h]: I personally think it's better to keep it as it is.

**Comment by sbutcher-arm** [comment]
@RonEld - Can you please backport the review feedback? #1779 and #1780 need the same fixes following the review of this PR.

**Comment by RonEld** [comment]
Done

**Comment by sbutcher-arm** [comment]
retest
